vernac/metasyntax.ml

```diff
@@ -1007,14 +944,90 @@ let check_useless_entry_types recvars mainvars etyps =
                   (Id.print x ++ str " is unbound in the notation.")
   | _ -> ()
 
-let interp_non_syntax_modifiers mods =
-  let set modif (only_parsing,only_printing,entry) = match modif with
-    | SetOnlyParsing -> Some (true,only_printing,entry)
-    | SetOnlyPrinting -> Some (only_parsing,true,entry)
-    | SetCustomEntry(entry,None) -> Some (only_parsing,only_printing,InCustomEntry entry)
-    | _ -> None
+type notation_main_data = {
```

Comment:
In general it is convenient to put this inside a module as to have better control over the scope of records projections, it also allows a more easy implementation of instances for the type such as `OrdType` etc...